server/service/software.go

```diff
@@ -147,16 +148,21 @@ func getSoftwareEndpoint(ctx context.Context, request interface{}, svc fleet.Ser
 }
 
 func (svc *Service) SoftwareByID(ctx context.Context, id uint, teamID *uint, includeCVEScores bool) (*fleet.Software, error) {
-	if err := svc.authz.Authorize(ctx, &fleet.AuthzSoftwareInventory{TeamID: teamID}, fleet.ActionRead); err != nil {
+	if err := svc.authz.Authorize(ctx, &fleet.Host{TeamID: teamID}, fleet.ActionList); err != nil {
```

Comment:
What' the reasoning around this change?

Reply:
I'm reverting to closer what is on main right now. (So I'm reverting my own earlier change.)
Basically, team roles have access to list hosts, so this method will return the software for the teams that the user has access to -- this is the recent change done by MDM team.